pkg/apprepo/apprepos_handler.go

```diff
@@ -52,33 +52,56 @@ type combinedClientset struct {
 	*kubernetes.Clientset
 }
 
-// AppRepositoriesHandler handles http requests for operating on app repositories
+// appRepositoriesHandler handles http requests for operating on app repositories
 // in Kubeapps, without exposing implementation details to 3rd party integrations.
-type AppRepositoriesHandler struct {
+type appRepositoriesHandler struct {
 	// The config set internally here cannot be used on its own as a valid
-	// token is required. Call-sites use ConfigForToken to obtain a valid
+	// token is required. Call-sites use configForToken to obtain a valid
 	// config with a specific token.
 	config rest.Config
 
 	// The namespace in which (currently) app repositories are created.
 	kubeappsNamespace string
 
-	// The Kubernetes client using the pod serviceaccount
-	svcKubeClient kubernetes.Interface
+	// clientset using the pod serviceaccount
+	svcClientset combinedClientsetInterface
 
 	// clientsetForConfig is a field on the struct only so it can be switched
-	// for a fake version when testing. NewAppRepositoryHandler sets it to the
+	// for a fake version when testing. NewAppRepositoryhandler sets it to the
 	// proper function below so that production code always has the real
 	// version (and since this is a private struct, external code cannot change
 	// the function).
 	clientsetForConfig func(*rest.Config) (combinedClientsetInterface, error)
+
+	clientset combinedClientsetInterface
+}
+
+type handler interface {
+	CreateAppRepository(appRepoBody io.ReadCloser, requestNamespace string) (*v1alpha1.AppRepository, error)
+	DeleteAppRepository(name, namespace string) error
+	GetNamespaces() ([]corev1.Namespace, error)
+	GetSecret(name, namespace string) (*corev1.Secret, error)
+	GetAppRepository(repoName, repoNamespace string) (*v1alpha1.AppRepository, error)
+}
+
+// AuthHandler exposes handler functionality as a user or the current serviceaccount
+type AuthHandler interface {
+	AsUser(token string) handler
+	AsSVC() handler
```

Comment:
Excellent, much more explicit while still being easy to use for call-sites. Thanks!

The only down-side I can think of here is that it now means that all the methods on the interface can be used with the service account, rather than just `GetNamespaces` and `GetAppRepository`. We could fix that by composing the interface and having AsSVC() return the more limited interface? Not sure. See what you think.

Reply:
That's what I wanted, if in the future we need to use this Handler with any the svc account or the user account we don't need to modify the library. Since it's quite explicit now, I don't think there is risk for a developer to misuse the method (nor a reviewer to miss it).